community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java

```diff
@@ -84,11 +84,13 @@ public State init( SessionStateMachine ctx, String clientName, Map<String,Object
                         }
                         catch ( AuthenticationException e )
                         {
-                            return error( ctx, new Neo4jError( e.status(), e.getMessage(), e ) );
+                            ctx.error( new Neo4jError( e.status(), e.getMessage() ));
+                            return halt( ctx );
```

Comment:
@pontusmelke correctly pointed out (in #7356) that I'm mistaken and it has to go to the ERROR state first (or an equivalent new state). So scratch this suggestion.